xilem_web/src/concurrent/async_repeat.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{future::Future, marker::PhantomData};
+
+use xilem_core::{DynMessage, Message, MessageResult, NoElement, View, ViewMarker};
+
+use crate::{context::MessageThunk, ViewCtx};
+
+pub fn async_repeat<M, F, H, State, Action, Fut>(
+    future_future: F,
+    on_event: H,
+) -> AsyncRepeat<F, H, M>
+where
+    F: Fn(MessageThunk) -> Fut + 'static,
```

Comment:
I'm not sure if we should expose this implementation detail of xilem_web, maybe wrap it? But I'm not sure if it's worth it to wrap.
I'm thinking about using `requestAnimationFrame` in the main loop to batch stuff, and throttle events and other messages before doing an actual `View::rebuild` but that may produce issues with input-latency (I'd need to test e.g. with https://www.vsynctester.com/ ) So not sure if worth it at least in the near-term, low-priority for me currently.

It's also somewhat dangerous, as currently the whole runtime of xilem_web relies on interior mutability, and when this is used in the same event loop frame, this'll lead to a crash because it's already borrowed (but I'm not sure if it's possible to provoke this though). I'm thinking about whether it's a good idea to just `queueMicrotask` (which I think is just `wasm_bindgen_futures::spawn_local`) in `MessageThunk::push_message` to defuse this risk.

Reply:
> I'm not sure if we should expose this implementation detail of xilem_web, maybe wrap it? 

There is now a `AsyncRepeatProxy` struct as wrapper.
BTW: What about the `xilem_core::MessageProxy`? It's not used in `xilem_web`. Isn't that something we're looking for here?

Reply:
I *guess* we could use it here, but it would make things more difficult with `web_sys` types, as it requires the `Send` bound, so I think having a separate proxy makes more sense and creates less headaches, unless we really want to support cross-web-worker stuff...